wp-admin/js/widgets/media-widgets.js

```diff
@@ -111,7 +111,9 @@ wp.mediaWidgets = ( function( $ ) {
 
 			// Re-render the preview when the attachment changes.
 			control.selectedAttachment = new wp.media.model.Attachment( { id: 0 } );
+			control.renderPreview = _.debounce( control.renderPreview );
```

Comment:
Great stuff, thanks for those links... so much to learn here.  So here is the `renderPreview` bit in action in the widget itself.  I've hacked the `alt` attribute in the `renderPreview` method to show how many times it is called.  Note that the number increments with each keystroke of the title which is triggering the `change` on `control.model`.  This is happening in `master` here too, so probably best to be tackled in another branch:

![refresh](https://cloud.githubusercontent.com/assets/22080/24769253/af464aac-1aba-11e7-8bc9-16a4ab8bb243.gif)